{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "OTU clustering \n",
    "====\n",
    "\n",
    "* Picrust Requires a closed otu file generated by qiime with genegenes database ( either verison 13.5 or 13.8 )\n",
    "* So below are the code which does the closed OTU picking \n",
    "* pick_closed_reference_otus.py makes a biom file\n",
    "* filter the biom files if required to remove the potential eukaryotic stuff and singltons\n",
    "* then add metadata to the files and upload it to phinch for graphical exploration\n",
    "\n",
    "----\n",
    "\n",
    "Tips\n",
    "----\n",
    "\n",
    "* run an open refernece otu picking also using pick_open_reference_otus.py\n",
    "* The resulting biom file will have *de novo* OTUs alongside green genes OTUs \n",
    "* Filter out those OTUs which are unassigned and add meta data into using\n",
    "\n",
    "----\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running Qiime for Closed OTU creation\n",
    "----\n",
    "```\n",
    "\n",
    "echo \"pick_otus:enable_rev_strand_match True\"  >> $PWD/otu_picking_params_97.txt\n",
    "echo \"pick_otus:similarity 0.97\" >> $PWD/otu_picking_params_97.txt\n",
    "\n",
    "workon QiimeVenv\n",
    "\n",
    "pick_closed_reference_otus.py -i $PWD/seqs.fna \\\n",
    "        -o $PWD/otu_clustering/ \\\n",
    "        -p $PWD/otu_picking_params_97.txt \\\n",
    "        -r $PWD/gg_13_5_otus/rep_set/97_otus.fasta \\\n",
    "        -a -O 20\n",
    "    \n",
    "deactivate\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Add metadata to the biom file to port it to Phinch\n",
    "----\n",
    "\n",
    "```\n",
    "biom add-metadata -i otu_table_mc2_w_tax.biom \\\n",
    "        -o otu_table_mc2_w_tax_and_metadata.json.biom \\\n",
    "        -m mapping_file.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tips:\n",
    "-----\n",
    "* Make duplicate copy of the #SampleID column in mapping_file.txt and rename the column as PhichID\n",
    "* Use this file biom add-metadata\n",
    "* This will help in labels in Phinch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filtering out the otu biom file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### filter choloplast , mt dna\n",
    "```\n",
    "filter_taxa_from_otu_table.py -i otu_table.biom \\\n",
    "        -o closed_ref_97_otu_table_no_euks.biom \\\n",
    "        -n cChloroplast,fmitochondria\n",
    "        \n",
    "        \n",
    "filter_taxa_from_otu_table.py -i closed_ref_97_otu_table_no_euks.biom \\\n",
    "        -o closed_ref_97_otu_table_no_euks_no_unassigned.biom \\\n",
    "        -n Unassigned\n",
    "```\n",
    "#### singlton removal\n",
    "```\n",
    "filter_otus_from_otu_table.py -i closed_ref_97_otu_table_no_euks_no_unassigned.biom \\\n",
    "        -o closed_ref_97_otu_table_filtered.biom  \\\n",
    "        -n 2\n",
    "```\n",
    "#### biom file conversion and change taxonomy to Consensus Lineage\n",
    "\n",
    "```\n",
    "biom convert -i closed_ref_97_otu_table_filtered.biom \\\n",
    "        -o closed_ref_97_otu_table_filtered.json --to-json \n",
    "        \n",
    "biom convert -i closed_ref_97_otu_table_filtered.biom \\\n",
    "        -o closed_ref_97_otu_table_filtered.tsv \\\n",
    "        --header-key taxonomy --to-tsv \n",
    "        \n",
    "        \n",
    "biom summarize-table -i closed_ref_97_otu_table_filtered.biom \\\n",
    "        -o closed_ref_97_otu_table_filtered.summary \n",
    "        \n",
    "sed s'/taxonomy/Consensus Lineage/' -i closed_ref_97_otu_table_filtered.tsv\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
